kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ingress/GatewayDefinition.java

```diff
@@ -8,31 +8,35 @@
 
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
+
 /**
- * Ingress definition for a single VirtualKafkaCluster, KafkaProxyIngress pair that:
+ * Gateway definition for a single VirtualKafkaCluster, KafkaProxyIngress pair that:
  * <ol>
- *     <li>declares the requirements of the ingress (e.g. how many identifying ports it requires)</li>
- *     <li>can instantiate IngressInstances</li>
+ *     <li>declares the requirements of the Gateway (e.g. how many identifying ports it requires)</li>
+ *     <li>can instantiate GatewayModel</li>
  * </ol>
+ * Corresponds to a single ingress item in the VirtualKafkaCluster spec.ingresses
  */
-public interface IngressDefinition {
+interface GatewayDefinition {
 
     /**
      * The raw resource that was translated into this definition
      * @return resource
      */
-    KafkaProxyIngress resource();
+    KafkaProxyIngress ingress();
 
     /**
-     * Create an instance of the Ingress with identifying ports allocated to it. Identifying meaning that
+     * Create a Gateway with identifying ports allocated to it. Identifying meaning that
      * the port on the container is expected to unambiguously identify which node the client is connecting to.
      * I.e. using a port-per-broker strategy at the proxy.
      *
-     * @param firstIdentifyingPort the first identifying port allocated to this ingress instance
-     * @param lastIdentifyingPort the last identifying port (inclusive) allocated to this ingress instance
-     * @return a non-null ingress instance
+     * @param firstIdentifyingPort the first identifying port allocated to this Gateway
+     * @param lastIdentifyingPort the last identifying port (inclusive) allocated to this Gateway
+     * @param sharedTLSPort if the definition requires a sharedTlsPort this must be non-null, else null
+     * @return a non-null GatewayModel
      */
-    IngressInstance createInstance(int firstIdentifyingPort, int lastIdentifyingPort);
+    GatewayModel createGateway(@Nullable Integer firstIdentifyingPort, @Nullable Integer lastIdentifyingPort, @Nullable Integer sharedTLSPort);
```

Comment:
Could think about introducing a PortAllocation object if it got any more complex